example/test_problem/Hydro/CCSN/python_script/convert_progenitor_form.py

```diff
@@ -0,0 +1,164 @@
+    headers = []
+    line_header = file.readline()
+    headers = [word.strip() for word in line_header.strip().split("  ") if word]
+
+    for i in range(len(headers)):
+        keys = ["cell outer radius", "cell density", "cell temperature", "cell outer velocity",
+                "cell Y_e", "cell pressure", "cell angular velocity"]
+
+    # find and read targeting variables (radius, density, temperature, radial velocity,
+    #                                    electron fraction, pressure, and angular velocity)
+    nvars = 7
+    TargetCols = np.empty(nvars, dtype=np.int)
+    for i in range(len(headers)):
+        keys = ["cell outer radius", "cell density", "cell temperature", "cell outer velocity",
+                "cell Y_e", "cell pressure", "cell angular velocity"]
+    TargetCols = [headers.index(key) for key in keys]
+
+    # GAMER-formatted array
+    GAMER_DATA_FORM = np.empty((0, nvars), dtype=np.float)
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    outer_radius = 0.0
+    outer_vel    = 0.0
+    for line in f:
+        if ('wind:' in line):
+            break
+
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line.split()[icol])
+            # convert cell outer variables to cell centered values (radius, velocity)
+            if idx == 0:
+                variables[idx], outer_radius = 0.5 * (outer_radius + variables[idx]), variables[idx]
+            if idx == 3:
+                variables[idx], outer_vel = 0.5 * (outer_vel + variables[idx]), variables[idx]
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+def convert_MESA(file):
+
+    # physical constants
+    R_sol = 6.977e10 # solar radius
+    M2CM  = 1e2      # meter to centimeter
+
+    # read the header
+    headers = []
+    f_info = ''
+    for i in range(5):
+        f.readline()
+    headers = f.readline().split()
+
+    # find and read targeting variables (radius, density, temperature, radial velocity,
+    #                                    electron fraction, pressure, and angular velocity)
+    nvars = 7
+    TargetCols = np.empty(nvars, dtype=np.int)
+    for i in range(len(headers)):
+        keys = ["logR", "density", "temperature", "velocity",
+                "ye", "pressure", "omega"]
+    TargetCols = [headers.index(key) for key in keys]
+
+    # GAMER-formatted array
+    GAMER_DATA_FORM = np.empty((0, nvars), dtype=np.float)
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    for line in f:
+
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line.split()[icol])
+            # unit conversion
+            if idx == 0:
+                variables[idx] = R_sol * np.power(10, variables[idx]) # unit conversion: radius
```

Comment:
Why add the comments twice? If the purpose is that L95 is not clear, you should also do so in `convert_WHS()`

Reply:
I removed these comments.